### PR TITLE
[core] Update Entities documentation

### PR DESCRIPTION
**Pull request by kbx81**
## Description:

Documentation for esphome/esphome#6885

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#6885

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/update/index.rst [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    :keywords: Updates, OTA, ESP32, ESP8266
+
+The `update` component allows your ESPHome devices to install updates from a remote source; instead of
+:doc:`"pushing" an update to your device </components/ota>`, the device can retrieve an update and install it on its own.
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    update:
+      - platform: ...
```
Comment (jesserockz) [inline components/update/index.rst:16]: name/id, icon, entity_category, device_class, disabled_by_default are all common update base items

File: components/update/http_request.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        source: http://example.com/manifest.yaml
+
+.. _update-configuration_variables:
+
+Configuration variables:
+------------------------
+
+- **id** (*Optional*, :ref:`config-id`): Manually specify the ID used for code generation.
+- **name** (*Optional*, string): The name for the update component.
+- **source** (**Required**, string): The URL of the YAML manifest file containing the firmware metadata.
```
Comment (jesserockz) [inline components/update/http_request.rst:30]: `update_interval` missing from here. Defaults to 6h

File: components/update/index.rst (hunk 1 above)
Comment (kbx81) [reply components/update/index.rst:16]: Done

File: components/update/http_request.rst (hunk 2 above)
Comment (kbx81) [reply components/update/http_request.rst:30]: And done

File: components/update/http_request.rst [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+The manifest consists of a JSON file structured as follows:
+
+.. code-block:: json
+
+    {
+      "name": "My ESPHome Project",
+      "version": "2024.6.1",
+      "builds": [
+        {
+          "chipFamily": "ESP32C3",
```
Comment (jesserockz) [inline components/update/http_request.rst]:
```suggestion
          "chipFamily": "ESP32-C3",
```

File: components/update/http_request.rst [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+- **source** (**Required**, string): The URL of the YAML manifest file containing the firmware metadata.
+- **update_interval** (*Optional*, :ref:`config-time`): The interval at which to check for updates. Defaults to 6 hours.
+- All other options from :ref:`Update <config-update>`.
+
+.. _update_http_request-manifest_format:
+
+Update Manifest Format
+----------------------
+
+The manifest consists of a JSON file structured as follows:
```
Comment (jesserockz) [inline components/update/http_request.rst]:
```suggestion
The manifest used by this component is the [ESP-Web-Tools manifest](https://github.com/esphome/esp-web-tools) with an extension in the ``ota`` block that is structured as follows:
```

**Review by jesserockz — APPROVED** [review]

**Pull request merged**